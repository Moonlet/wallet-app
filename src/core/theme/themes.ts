export enum Theme {
    dark = 'dark',
    light = 'light'
}
