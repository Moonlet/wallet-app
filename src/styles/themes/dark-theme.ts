import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
