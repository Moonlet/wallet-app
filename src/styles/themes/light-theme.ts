import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
