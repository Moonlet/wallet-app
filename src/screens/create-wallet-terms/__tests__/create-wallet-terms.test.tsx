import React from 'react';
import {
    CreateWalletTermsScreenComponent,
    IProps,
    navigationOptions
} from '../create-wallet-terms';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: IProps = {
    // @ts-ignore
    navigation: {
        navigate: jest.fn()
    },
    styles: styleProvider(darkTheme)
};

describe('creat wallet terms screen component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<CreateWalletTermsScreenComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('navigates correctly', () => {
        const wrapper = shallow(<CreateWalletTermsScreenComponent {...props} />);

        wrapper.find('[testID="button-accept"]').simulate('Press');
        expect(props.navigation.navigate).toHaveBeenCalledWith('CreateWalletMnemonic');

        wrapper.find('[testID="button-tos"]').simulate('Press');
        expect(props.navigation.navigate).toHaveBeenCalledWith('Tos');

        wrapper.find('[testID="button-privacy-policy"]').simulate('Press');
        expect(props.navigation.navigate).toHaveBeenCalledWith('PrivacyPolicy');
    });

    it('sets correct navigation options', () => {
        const navigationProp = { navigation: { state: { params: { goBack: jest.fn() } } } };
        const options = navigationOptions(navigationProp);
        expect(options).toMatchSnapshot();
        expect(options.headerLeft()).toMatchSnapshot();
    });

    it('does not have a back button if no goBack param is set', () => {
        const navigationProp = { navigation: {} };
        const options = navigationOptions(navigationProp);
        expect(options.headerLeft()).toBe(null);
    });
});
