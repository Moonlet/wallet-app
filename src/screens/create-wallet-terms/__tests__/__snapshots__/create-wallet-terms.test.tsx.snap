// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creat wallet terms screen component renders correctly 1`] = `
"<View style={{...}}>
  <View style={{...}}>
    <withTheme(TextComponent) darker={true} style={{...}}>
      Make sure you review our Privacy policy and Terms of service
    </withTheme(TextComponent)>
    <View style={{...}}>
      <Image source={{...}} />
    </View>
  </View>
  <View style={{...}}>
    <TouchableOpacity testID=\\"button-tos\\" style={{...}} onPress={[Function: onPress]} activeOpacity={0.2}>
      <withTheme(TextComponent)>
        Terms of service
      </withTheme(TextComponent)>
      <View style={{...}}>
        <Icon name=\\"arrow-right-1\\" size={16} style={{...}} allowFontScaling={false} />
      </View>
    </TouchableOpacity>
    <View style={{...}} />
    <TouchableOpacity testID=\\"button-privacy-policy\\" style={{...}} onPress={[Function: onPress]} activeOpacity={0.2}>
      <withTheme(TextComponent)>
        Privacy policy
      </withTheme(TextComponent)>
      <View style={{...}}>
        <Icon name=\\"arrow-right-1\\" size={16} style={{...}} allowFontScaling={false} />
      </View>
    </TouchableOpacity>
    <View style={{...}} />
    <withTheme(ButtonComponent) testID=\\"button-accept\\" style={{...}} primary={true} onPress={[Function: onPress]}>
      Accept
    </withTheme(ButtonComponent)>
  </View>
</View>"
`;

exports[`creat wallet terms screen component sets correct navigation options 1`] = `
Object {
  "headerLeft": [Function],
  "title": "Create",
}
`;

exports[`creat wallet terms screen component sets correct navigation options 2`] = `
<withTheme(HeaderLeftComponent)
  icon="arrow-left-1"
  onPress={[Function]}
  text="Back"
/>
`;
